lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2478,13 +2478,44 @@ LogicalResult ConnectOp::verify() {
   return success();
 }
 
+// type(%dest) ^(, type(%src))
+static ParseResult parseOptionalConnectOperandTypes(OpAsmParser &parser,
```

Comment:
These are great, thanks!  Bet makes test churn a lot smaller too :grin:.

Not big deal, but 1)Might be good to push these to dedicated file like we do for HW (and in Support), especially since they aren't very FIRRTL-specific (other than the naming re:connect/src/dest)--it's basically custom parsing/printing for two types where second may be same as first.  2)Is it okay for these to be `static` -- I guess they're never used outside this file? (just curious)

Reply:
Move implementations to `include/circt/Support/CustomDirectiveImpl.h`. 

> 2)Is it okay for these to be static -- I guess they're never used outside this file? (just curious)

It's ok since they are used by FIRRTLOps.inc file which is included by this file. We have made custom directive helpers static conventionally.